### ISSUE TITLE
Really low resolution after cropped with Resizable Custom crop

### ISSUE THREAD

**Issue by Shasoosh** [gekitz/GKImagePicker#7]
Just found a major bug
I just cropped a big image of a text document and the text is unreadable. If I go to my camera roll and zoom in on the text it's crisp and clear.
I've found out that a 2000x2000 image is now 600x500. (and the size of it haven't changed, I only cropped one line of pixels)
You can repredouce the bug by setting the crop points in Resizable Custom crop on a ZOOMED-OUT picture (this is where the bug is) If you'll zoom-in on the picture and then crop it it will be fine.
Thanks

**Comment by gekitz** [comment]
Well I never had any problems with that. I suggest you add the image to this issue and I'll take a look as soon as I've some time to do it.

**Comment by Shasoosh** [comment]
Original picture: 
Resizable Custom Crop -> moved the points around and cropped while the picture is still zoomed out: 
Resizable Custom Crop -> Here I zoomed in and then played around with the points and cropped:

**Comment by Shasoosh** [comment]
I found the above bug 10 min before posting my app to the appstore and spent most of the weekend on this trying to find a solution. The bug accures in all crop modes, not just Resizable Custom Crop like I wrote above. If you are zoomed out of the picture you won't get the real image resolution after you crop it. I suspect that the all scaling of the image after the crop is some how wrong. 
@gekitz I'd really appreciate it if you could share your thoughts about this. With your understand of the code even an idea or an hint can help. Thanks

**Comment by gekitz** [comment]
tried to reproduce your problem, but everything looks quite good on my iPhone 5. Which cropSize do you use?

**Comment by Shasoosh** [comment]
My problem exists in the original demo I downloaded from github without changing any setting. Download to your phone the large  picture I posted above. With the gkimagepicker demo select realizable crop, and crop one line of text. If the picture is zoomed out before the crop then the line of text would be unreadable after crop. This happens on iOS 5, 6 under the Xcode simulator. I can upload a video if you think it would help

**Comment by gekitz** [comment]
this really weird, I used the latest version and everything seems to work quite good. 
That's indeed really weird!

**Comment by Shasoosh** [comment]
I'm definitely uploading a video clip to show you my steps, until then, do
you mind testing something out ?

Take any big picture , mine is at
Resolution 3200x2000, I cropped exactly just the left side of it so i
should be getting a 1600x2000 picture, right? Look on the dimensions of the
new cropped picture :
https://www.dropbox.com/s/2zr3ip2fmq9dv6o/Photo%2028-01-13%2021%2029%2026.png

The resolution of the cropped image is 308x376. Very very low res. If you
repeat this process, do you get a resolution of 1600x2000? (the resolution
you should get...)

**Comment by gekitz** [comment]
made a branch bug/scaling where I tweaked a view things. Please check if that helps.

**Comment by Shasoosh** [comment]
Ok,  dowloaded it from the new branch and fter cropping just half of  a 3000x2000 image the result image is still 300x400 instead of 1500x2000. The picture loses almost 90 % of its resolution.

**Comment by oded244** [comment]
I joined Github to find a descent cropping tool and I'm facing the same issues the OP is facing. Cropping the picture lowers the image resolution drastically and makes it unusable. Compared to other cropping tools I've found GKImagePicker is very light weight in terms of memory  usage so I'd also +1 to see this fixed.

**Comment by ghost** [comment]
Same problem, any idea ?
I think the cropped area does not picked on the original image but on the shown image.

**Comment by gekitz** [comment]
hey guys,

please check out the experimental/better-crop-image-quality branch of the project. 
I think I could fix the issue you, do me a favor and verify it, you should see a better crop quality if you use "Custom Crop".

Would be happy to get some feedback.

**Comment by Shasoosh** [comment]
Yes. Custom Crop gives a good resolution cropped image. Just apply the changes you made there to the other crop methods. Good job!

**Comment by Shasoosh** [comment]
There's still problems if I crop a big picture to approximately the same size ~ the result will be a blurry low res picture. Cropping a small area from a big picture works great though.

**Comment by Abderezai** [comment]
I still see this issue, at least for the resizable mode. It appears that you are cropping the actual size and resolution of the display, so the best resolution is approximately 320 width (max width of an iphone) and whatever the corresponding hight would be.

**Comment by convey-mane** [comment]
as @gekitz was closing issue https://github.com/gekitz/GKImagePicker/issues/11 with https://github.com/gekitz/GKImagePicker/commit/2a91f31a90ad2dca02b58395a2604095f8c96fbf - still leaving the problem with "Resizeable Custom Crop" unsolved - he therefore created a new issue #12 delegating it to @pathonhauser, who implemented that "Resizeable" part. There I myself left some comments - also today.

Maybe that one and this here could be "merged"?

**Comment by convey-mane** [comment]
(ah, OK, there are references inserted in the mentioned issues - sorry, I'm very new here ;-)  ).
